File: gcd/src/main/java/local/example/outcome/OutcomeResource.java
package local.example.outcome;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;

import java.util.Collections;
import java.util.LinkedHashMap;
import java.util.Set;

@Path(value = "/outcome")
public class OutcomeResource {

    private final Set<Gcd> gcdSet = Collections.newSetFromMap(
            Collections.synchronizedMap(
                    new LinkedHashMap<>()
            )
    );

    @GET
    public String outcome() {
        return "-- outcome feedback --";
    }

    @POST
    @Produces(value = MediaType.APPLICATION_JSON)
    @Consumes(value = MediaType.APPLICATION_JSON)
    public Set<Gcd> compute(Gcd gcd) {
        gcd.setGcd();
        gcdSet.add(gcd);
        return gcdSet;
    }
}

File: average/src/main/kotlin/local/example/outcome/OutcomeResource.kt
package local.example.outcome

import local.example.outcome.model.Average
import java.util.*
import javax.ws.rs.*
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/outcome")
class OutcomeResource {

    private val averages = Collections.newSetFromMap(
        Collections.synchronizedMap(
            LinkedHashMap<Average, Boolean>()
        )
    )

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    fun read(): Response? {
        return if (averages.isEmpty()) Response.ok().build() else Response.ok(averages).build()
    }

    @POST
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    fun compute(average: Average): Set<Average?>? {
        average.setAverage()
        averages.add(average)
        return averages
    }
}
